### PR TITLE
Allow for endpoint on article links to be used for query vars or not at all

### PR DESCRIPTION
**Pull request by AramZS**
This provides a useful solution to #34

### CODE REVIEW

**Comment by whyisjake** [comment]
I like this solution, as a I think as a default we should leave the endpoint open. Since there is a canonical URL, we don't need to worry about any SEO implications. 

I am going to do a little testing, and then will likely merge.

**Comment by AramZS** [comment]
@whyisjake great!

**Comment by dashaluna** [comment]
Cross posting comment here from #39 as I don't know which branch will make it in first.
I have noticed some [WP PHP Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/) issues. Do you want me to clean them up in a commit or add comments to this PR and you can do it? I don't mind either way, but would be great to fix them before merge :) Sorry for nitpicking!

**Comment by whyisjake** [comment]
@dashaluna, I went ahead with the merge. Wanted to get this out, but we should still update to the coding standards. I'll take a stab at cleaning things up, but a PR is welcome too. :)

**Comment by whyisjake** [comment]
@AramZS, this was a huge pull. thanks for the great work!

**Comment by AramZS** [comment]
@whyisjake Much thanks!

File: simple-fb-instant-articles.php [hunk 1]
```diff
@@ -86,11 +84,19 @@ function add_actions() {
...
 
-		if ( false !== get_query_var( $this->endpoint, false ) ) {
+		if ( false !== get_query_var( $this->endpoint, false ) && $this->is_redirectable_endpoint() ) {
 			add_action( 'template_redirect', array( $this, 'template_redirect' ) );
 		}
 	}
 
+	function is_redirectable_endpoint(){
+		if ('' === $this->endpoint || 0 == strpos($this->endpoint, '?') ){
+			return false;
```
Comment (whyisjake) [inline simple-fb-instant-articles.php:94]: Are these vals swapped?

Comment (AramZS) [inline simple-fb-instant-articles.php:94]: @whyisjake I don't think so, If a dev doesn't want an endpoint they can filter it to empty or they can attach a campaign tag to the end, in which case we wouldn't want to create a new endpoint. This works for that, right?

**Pull request merged**